### PR TITLE
[14.0][ADD] shopinvader_customer_activity

### PR DESCRIPTION
**Pull request by ivantodorovich**
Log the customer's shop activity dates.

This module will add 2 new fields to the `shopinvader.partner` model and views:

* First Activity on
* Last Activity on

### CODE REVIEW

**Review by SilvioC2C — APPROVED** [review]
LGTM

File: shopinvader_customer_activity_date/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
```
Comment (simahawk) [inline shopinvader_customer_activity_date/__manifest__.py]: I'd remove the `_date` from this module name to ease further extension

File: shopinvader_customer_activity_date/services/service.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+from odoo.addons.component.core import AbstractComponent
+
+
+class BaseShopinvaderService(AbstractComponent):
+    _inherit = "base.shopinvader.service"
+
+    def dispatch(self, method_name, *args, params=None):
+        res = super().dispatch(method_name, *args, params=params)
+        if self.invader_partner_user:
```
Comment (simahawk) [inline shopinvader_customer_activity_date/services/service.py]: this triggers a write on all calls. I would enable/disable it via flags on  the backend

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader_customer_activity_date/services/service.py (hunk 2 above)
Comment (ivantodorovich) [reply shopinvader_customer_activity_date/services/service.py]: added a roadmap entry to make it configurable

File: shopinvader_customer_activity_date/__manifest__.py (hunk 1 above)
Comment (ivantodorovich) [reply shopinvader_customer_activity_date/__manifest__.py]: agree, module renamed!

**Review by simahawk — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Comment by ivantodorovich** [comment]
> @ivantodorovich I'd say the tech user should be skipped completely here. There's no need to track its activity IMO.

Normally the tech user wouldn't have an associated `shopinvader.partner`, so this wouldn't apply to it. There's nothing to track.
It only applies to requests made by "authenticated" shop partners (so, customers essentially)

**Pull request merged**